### PR TITLE
Port make-lightgbm

### PR DESCRIPTION
**Pull request by shengwangsw**
Port `make-lightgbm`

### CODE REVIEW

**Comment by shengwangsw** [comment]
I can't add @AlbertoEAF to this code review, I will appreciate if someone can do it for me :smile:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Author and maintainer: Alberto Ferreira alberto.ferreira@feeczai.com.
```
Comment (ptraca) [inline CONTRIBUTING.md]: @AlbertoEAF Should this be here, anyway there is typo on your email.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+Patches are welcome.
+
+All this code is intellectual property of Feedzai under the provisioned licence.
```
Comment (ptraca) [inline CONTRIBUTING.md]: I think this license needs to be reviewed, it's not correct (we are open-sourcing this)

**Comment by ptraca** [comment]
@shenggwang I've finished my review, please see comments.

File: CONTRIBUTING.md (hunk 1 above)
Comment (shengwangsw) [reply CONTRIBUTING.md]: I'll change it into the files

File: CONTRIBUTING.md (hunk 2 above)
Comment (shengwangsw) [reply CONTRIBUTING.md]: Should we use apache license? [like this one ](https://github.com/feedzai/feedzai-openml-java/blob/master/LICENSE)

Comment (shengwangsw) [reply CONTRIBUTING.md]: João Sousa just told me, the right License is under `/resource` folder

**Comment by shengwangsw** [comment]
I think now it is ready to go.
@JPDSousa @ptraca

**Comment by JPDSousa** [comment]
> I can't add @AlbertoEAF to this code review, I will appreciate if someone can do it for me 😄

I think it's because he's not a part of the openml-maintainers. Just ping him directly

File: CONTRIBUTING.md (hunk 1 above)
Comment (AlbertoEAF) [reply CONTRIBUTING.md]: feeczai lol shame.

File: CONTRIBUTING.md (hunk 2 above)
Comment (AlbertoEAF) [reply CONTRIBUTING.md]: But under /licence that's for the OpenMP, not our tool to build LightGBM @shenggwang @JPDSousa @ptraca .

I don't know which licence to use, you guys decide.

**Comment by AlbertoEAF** [comment]
From our conversation I thought we should not add the output build/ folder to the repo. The repo is just the tool and does not include its outputs. And then we include this whole repo as submodule in the provider and the person can check the instructions to produce the lightgbm in the desired version, which will very rarely used. 

 Right @ptraca @JPDSousa @shenggwang ?

File: resources/libopenmp.licence [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Copyright (C) 2005, 2007, 2008, 2009 Free Software Foundation, Inc.
```
Comment (JPDSousa) [inline resources/libopenmp.licence:1]: This file is a duplicate of the one in `build/`

**Review by JPDSousa — CHANGES_REQUESTED** [review]

**Comment by JPDSousa** [comment]
> From our conversation I thought we should not add the output build/ folder to the repo. The repo is just the tool and does not include its outputs. And then we include this whole repo as submodule in the provider and the person can check the instructions to produce the lightgbm in the desired version, which will very rarely used.
> 
> Right @ptraca @JPDSousa @shenggwang ?

I guess that makes sense 🤔

**Comment by shengwangsw** [comment]
> From our conversation I thought we should not add the output build/ folder to the repo. The repo is just the tool and does not include its outputs. And then we include this whole repo as submodule in the provider and the person can check the instructions to produce the lightgbm in the desired version, which will very rarely used.
> 
> Right @ptraca @JPDSousa @shenggwang ?

I thought that it is supposed to create the build folder on make-lightgbm project, and on the openml-lightgbm we use the build folder as submodule.

@JPDSousa @ptraca correct me if I'm wrong.

File: resources/libopenmp.licence (hunk 3 above)
Comment (AlbertoEAF) [reply resources/libopenmp.licence:1]: We should not commit the build/ but this instead, that whole folder is generated from existing resources in the other folders, such as this one.

Comment (shengwangsw) [reply resources/libopenmp.licence:1]: build/  removed

**Review by JPDSousa — APPROVED** [review]

**Comment by shenggwang** [comment]
Port make-lightgbm

**Pull request merged**